internal/server/data/provideruser.go

```diff
@@ -296,40 +297,31 @@ func ProvisionProviderUser(tx WriteTxn, user *models.ProviderUser) error {
 	return insert(tx, (*providerUserTable)(user))
 }
 
-func SyncProviderUser(ctx context.Context, tx WriteTxn, user *models.Identity, provider *models.Provider, oidcClient providers.OIDCClient) error {
-	providerUser, err := GetProviderUser(tx, provider.ID, user.ID)
-	if err != nil {
-		return err
-	}
-
-	accessToken, expiry, err := oidcClient.RefreshAccessToken(ctx, providerUser)
+func SyncProviderUser(ctx context.Context, tx WriteTxn, user *models.ProviderUser, oidcClient providers.OIDCClient) ([]models.Group, error) {
```

Comment:
nit for later: we should probably move this out of `data` since it's more about making an external call to the identity provider and calling `data` functions

Reply:
I wonder if we need to store the provider user in the database at all since it's mostly functioning a cache for the user's OIDC access token. Perhaps we can store it in Redis instead? Expiry can simply be a setting a TTL on the Redis key and when a key isn't found, call `oidcClient.RefreshAccessToken`

Reply:
I believe we need the provider user in the database to be able to identify which IDPs provide a particular user. It's much easier to do that with a SQL `JOIN` clause, than having to query multiple data sources and doing the join in the application.